src/test/injecture-test.js

```diff
@@ -215,4 +215,137 @@ describe('Injecture', function() {
 
     done();
   });
+
+
+  it('can return an instance from an interface', (done) => {
+
+    class ClassN {}
+    injecture.registerClass(ClassN, {
+      interfaces: ['baz']
+    });
+
+    const classN = injecture.getInstanceByInterface('baz');
+    assert.ok(classN instanceof ClassN);
+
+    done();
+  });
+
+  it('when there is no reducer, getInstanceByInterface returns the first registered', done => {
+    class ClassO {}
+    injecture.registerClass(ClassO, {
+      interfaces: ['beep'],
+      singleton: true,
+    });
+
+    const classO = injecture.getInstanceByInterface('beep');
+    assert.ok(classO);
+    assert.ok(classO instanceof ClassO);
+
+    // This should POSSIBLY throw. Probably not. I can't think of a valid use case
+    // for having two singletons that implement the same interfaces (in the same
+    // thread/runtime at least), but I'm 67% sure that one exists.
+    injecture.registerClass(class ClassP {}, {
+      interfaces: ['beep'],
+      singleton: true,
+    });
+
+    const keys = injecture.getKeysByInterface('beep');
+    assert.equal(keys.length, 2, 'prove that interface beep has two classes');
+
+
+    // This should DEFINITELY throw, or at least produce a warning. Something is
+    // fishy when we are asking for a singleton by interface and there are
+    // multiple candidates.
+    const secondClassO = injecture.getInstanceByInterface('beep');
+    assert.ok(secondClassO);
+    assert.ok(secondClassO instanceof ClassO);
+
+    // TODO: spy console.warn to ensure it was called
+
+    done();
+  });
+
+
+  it('getInstanceByInterface will create an instance from the key the reducer returns', done => {
+
+    class ClassQ {
+      get() { return 'nike'}
+    }
+    class ClassR {
+      get() { return 'rebok'}
+    }
+
+    injecture.registerClass(ClassQ, {
+      interfaces: ['shoe']
+    });
+
+    injecture.registerClass(ClassR, {
+      interfaces: ['shoe']
+    });
+
+    let classToFilter = 'ClassR'
+
+    injecture.addInterfaceReducers({
```

Comment:
INTERESTING QUESTION: Should this filter on the class keys/names as you are here? Or should it filter on the instances? Like, get all instances of any classes that fit, then filter on those instances instead of the classes. What other knobs do you see giving devs with this mechanism?

Reply:
Being too literal bruh, gotta focus on the message. Forget "filters".

What you're demonstrating in your test is that reducers receive keys as their input. My question is: what other interesting things might (or could) they receive as arguments?

Reply:
rainbows

Reply:
Related: Might we have different types of reducers? key reducers vs interface reducers vs ???

Reply:
that was supposed to say "instance reducers"